### ISSUE TITLE
Messy component.

### ISSUE THREAD

**Issue by ronanyeah** [shahenazmonia/tech-resource#54]
https://github.com/shahenazmonia/tech-resource/blob/master/src/frontEnd/components/myTable.jsx#L66

This should be two buttons, like:
```
this.props.trStatus === elem.id
  ? <button>
        // editing stuff
     </button>
  ? <button>
        // update button
     </button>
```

**Comment by ronanyeah** [comment]
Also the freeze action is unnecessary, when you call updateResource, update resource should call the freeze resource action before it sends off the fetch request.

**Comment by ronanyeah** [comment]
In fact I think you should create a new component that is called `editingRow` that is returned from the table map if the editing ID matches. Pass in the resource values as props and use them as state but just change state of the row within the component. Then fire off the updateResource action from inside it. I know this sounds mega complicated but give it a shot.

**Comment by mhmdshorafa** [comment]
https://github.com/shahenazmonia/tech-resource/tree/editRowComponent

if you can check this branch i don't want to merge it before being sure if i did it the correct way

**Comment by ronanyeah** [comment]
@mhmdshorafa Yeah you're getting there. Use the state of the component to manage the changes of the text areas, just like you are doing with the form. Except here the initial value of the state will come from teh props, [here](https://github.com/shahenazmonia/tech-resource/blob/editRowComponent/src/frontEnd/components/editingRow.jsx#L6) in the constructor. This editingRow will basically act like a little form. If you click save it does the update.

**Comment by mhmdshorafa** [comment]
if the state necessary ? because i user refs to get the text from the td .. and the update works fine now ... if there is something i miss for using the state in this component please tell me .. thnx
https://github.com/shahenazmonia/tech-resource/blob/master/src/frontEnd/components/editingRow.jsx#L19
https://github.com/shahenazmonia/tech-resource/blob/master/src/frontEnd/components/editingRow.jsx#L22

**Comment by ronanyeah** [comment]
I'll give it a test but I'd prefer if you didn't use refs. You are also mutating the `let` variables you are creating, which makes the logic messy to follow, and you are using them in a way that state is supposed to be used for.

**Comment by ronanyeah** [comment]
https://facebook.github.io/react/docs/refs-and-the-dom.html

**Comment by ronanyeah** [comment]
Your use of refs is also probably the reason for the errors in the console.
`Warning: A component is 'contentEditable' and contains 'children' managed by React.`
Errors like that won't be allowed into master next week... :smile:

**Comment by mhmdshorafa** [comment]
i am so exited for the next week .. i'll remove the refs and i will use the state instead .. i'll attach the commit ..

**Comment by mhmdshorafa** [comment]
i'm afraid that i can't use onChange or onInput methods for td .. i think that i have to put an input inside the td .. do u think there is any other solution ?

**Comment by ronanyeah** [comment]
Yeah stick an input in there.

**Comment by ronanyeah** [comment]
Check if the errors go away too, hopefully they do.

**Comment by mhmdshorafa** [comment]
https://github.com/shahenazmonia/tech-resource/commit/290898e96f74a9508a03f2ea8af3f084da01ed4e
i added a new component to add the input to td 
and used the trStatus reducer to store the new values 
the error is gone

**Comment by ronanyeah** [comment]
Looking ok. Why is the state an object again? A number and null was working fine.
Get rid of the new component, just put everything in the editRow component. Also there's a callback in one of your functions for some reason, get rid of it.

**Comment by ronanyeah** [comment]
Also just put everything in master, the branches are confusing since we aren't doing pull requests.

**Comment by mhmdshorafa** [comment]
03845caba6cc64a32ded93f7796be9d961598cfe
